src/com/jwetherell/algorithms/strings/StringFunctions.java

```diff
@@ -14,13 +14,23 @@ public class StringFunctions {
 
     public static final String reverseWithStringConcat(String string) {
         String output = new String();
+
+        if(string == null || string.IsEmpty()) {
```

Comment:
This if statement guard protects against invalid memory accesses by checking if the arg String object is empty or equal to null.